### PR TITLE
Add option to preserve counts in Map.interp_to_geom()

### PR DESCRIPTION
**Pull request by LauraOlivera**
**Description**

This pull request adds an option `preserve_counts` to the method `Map.interp_to_geom()`. It is then used in the data reduction step of a `MapDataset` for the background, in case the input is a `Map`.

It also makes it optional to have an exposure map for the `edisp` and `psf` in a dataset when saving it to a file (this was missing from PR [#2990](https://github.com/gammapy/gammapy/pull/2990))

### CODE REVIEW

File: gammapy/maps/core.py [hunk 1]
```diff
@@ -776,11 +780,16 @@ def interp_to_geom(self, geom, **kwargs):
         if self.data.dtype == bool:
             kwargs.setdefault("interp", 0)
 
-        data = self.interp_by_coord(coords, **kwargs)
+        if preserve_counts:
+            old_map_copy = self.copy()
+            old_map_copy.data /= self.geom.solid_angle().to_value("deg2")
```
Comment (adonath) [inline gammapy/maps/core.py:785]: I just realised, that the bins size in energy could change as well. So we probably need a more general solution like:
```python3
bin_volume = self.geom.bin_volume.value
old_map_copy.data /= bin_volume

data *= geom.bin_volume.to_value(bin_volume.unit)

```

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @LauraOlivera, I have left one inline comment...otherwise looks good to me.

File: gammapy/maps/core.py (hunk 1 above)
Comment (adonath) [reply gammapy/maps/core.py:785]: Conclusion from the dev meeting today: keep the implementation, but add a check on the equality of the energy axis. This simply works like `energy_axis_1 == energy_axis_2`...

Comment (LauraOlivera) [reply gammapy/maps/core.py:788]: done! :+1:

**Review by adonath — APPROVED** [review]
Thanks a lot @LauraOlivera! No further comments from my side...

**Pull request merged**